pkg/flow/internal/controller/loader.go

```diff
@@ -123,7 +131,7 @@ func NewLoader(opts LoaderOptions) *Loader {
 // to expose values of other components.
 //
 // declareBlocks are pieces of config that can be used as templates to instantiate custom components.
-func (l *Loader) Apply(args map[string]any, componentBlocks []*ast.BlockStmt, configBlocks []*ast.BlockStmt, declareBlocks []*ast.BlockStmt) diag.Diagnostics {
+func (l *Loader) Apply(args map[string]any, componentBlocks []*ast.BlockStmt, configBlocks []*ast.BlockStmt, declareBlocks []*ast.BlockStmt, options LoadOptions) diag.Diagnostics {
```

Comment:
The parameter list here is getting out of hand. Can you move all of the parameters to LoadOptions?

Reply:
There are a few ways we can go if we put these params in the options.
Currently the functions are as follow:
```
func (c *module) LoadBody(body ast.Body, args map[string]any, options ApplyOptions) error
func (f *Flow) loadSource(source *Source, args map[string]any, options controller.ApplyOptions) error
func (l *Loader) Apply(args map[string]any, componentBlocks []*ast.BlockStmt, configBlocks []*ast.BlockStmt, declareBlocks []*ast.BlockStmt, options LoadOptions) diag.Diagnostics 
```
One way to change it could be:
```
func (c *module) LoadBody(body ast.Body, args map[string]any, customComponentRegistry *CustomComponentRegistry) error
func (f *Flow) loadSource(source *Source, args map[string]any, customComponentRegistry *CustomComponentRegistry)
func (l *Loader) Apply(options ApplyOptions)
```
What do you think?

Reply:
That's a lot better, I'm in favor 👍